src/plone/restapi/services/users/get.py

```diff
@@ -222,7 +223,12 @@ def _get_user_id(self):
     def render(self):
         if len(self.params) == 1:
             # Retrieve the user portrait
-            portrait = self.portal_membership.getPersonalPortrait(self.params[0])
```

Comment:
Here be dragons.

There is a long (>15y) history of patching `MembershipTool.getPersonalPortrait` in several addons dealing with it. Membrane did/does, pas.plugins.ldap as well, probably others too.

Before you merge this please check if this works with a recent version of `pas.plugins.ldap` or similar (should, but better safe than sorry).  I am also not sure if the meta_type assumption  always holds.

Reply:
So if I understand your comment @jensens you're suggesting it would be dangerous to not use `getPersonalPortrait`?

Because other addon's may patch it to give a different result, so by not using it we could be missing out on those changes?

I can see in pas.plugins.ldap it is patching it to basically solve the issue we are seeing here: https://github.com/collective/pas.plugins.ldap/blob/e1f8da6ab300bed50c779febac9036930b2cb91f/src/pas/plugins/ldap/monkey.py#L58-L86

Unfortunately, `portal_membership.getPersonalPortrait` doesn't return a portrait if the user id is already sanitised.

Perhaps the better solution would be to 'de-sanitize' the user id before passing it to `getPersonalPortrait`?

Reply:
Exact, yes. And I am not sure if the meta_type is always the one you expect.

Reply:
ok thanks for the feedback. I'll take another look at the meta_type check, I'm sure there is a better way to determine if the users portrait is unset.

Reply:
@jensens I have refactored the code.

So now both the code that returns the portrait url and fetches the portrait will use getPersonalPortrait.

When fetching from the @portrait endpoint, it is expecting a 'safe id' so is calling `decleanId` from pas before calling `getPersonalPortrait`.

I don't really have an effective setup to test pas.ldap (and certainly not with custom portrait code), but as we are using the `getPersonalPortrait` function as intended I can't see it being an issue there due to this change.

Reply:
Yes, this will do it. Great work. It is always good to see others getting their fingers dirty with our membership code ;-)